src/app/features/notification/model/notification-config.ts

```diff
@@ -61,7 +60,15 @@ export class NotificationRule {
   @DatabaseField() entityType: string;
 
   /** (for "entity_change" notifications only): conditions which changes cause notifications */
-  @DatabaseField() conditions: DataFilter<any>;
+  @DatabaseField() conditions: NotificationCondition[];
+}
+
+export class NotificationCondition {
+  @DatabaseField() entityTypeField: string;
+
+  @DatabaseField() operator: string;
+
+  @DatabaseField() condition: string;
 }
```

Comment:
I would indeed like the "conditions" to be of type `DataFilter`, for example

```json
{
  "user": { "$eq": "my-user-id" },
  "status": { "$in": ["NEW", "TO_BE_APPROVED" ] }
}
```

This makes it identical/compatible with our other filters. Although you will need to parse it differently to display one condition component per object property then.

Reply:
I will update the notification condition logic accordingly.